# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/PBackground.ipdl
# Commit: 3eb8fdd0ba6d
# Full Hash: 3eb8fdd0ba6d2d921abe85086539b15d1e855686
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-12 21:30:33
# Regressor Bug: 1351231
# File Overlap Count: 2
# Description:
#   Bug 1351231 - PFetch protocol declaration and implementation. r=dom-worker-reviewers,jesup
#   
#   Depends on D138813
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142436
# ==============================================================================

diff -r a1fcf22a2a0e -r 3eb8fdd0ba6d ipc/glue/PBackground.ipdl
--- a/ipc/glue/PBackground.ipdl	Thu Jan 12 17:24:27 2023 +0000
+++ b/ipc/glue/PBackground.ipdl	Thu Jan 12 17:24:28 2023 +0000
@@ -53,6 +53,7 @@
 include protocol PRemoteDecoderManager;
 include protocol PWebSocketConnection;
 include protocol PWebTransport;
+include protocol PFetch;
 
 include DOMTypes;
 include IPCBlob;
@@ -131,6 +132,7 @@
   manages PUDPSocket;
   manages PVerifySSLServerCert;
   manages PVsync;
+  manages PFetch;
 
 parent:
   // Only called at startup during mochitests to check the basic infrastructure.
@@ -321,6 +323,8 @@
 
   async PIPCClientCerts();
 
+  async PFetch();
+
 child:
   async PCache();
   async PCacheStreamControl();
